### PR TITLE
ci: Fix it + Lint changes

### PR DESCRIPTION
**Pull request by Alc-Alc**
- Update CI actions to fix `awesome-lint`
- Removed duplicate links.
- Plugin links were the most duplicated ones, I reworded it to remove the discrepancy while still "emphasizing" the plugin enabled repos.
- Reddit link 403s, idk why, we can ignore that?

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,28 +1,30 @@
 name: CI
 
 on:
-  - push
-  - pull_request
+  pull_request:
```
Comment (sobolevn) [inline .github/workflows/ci.yml:5]:
```suggestion
  workflow_dispatch:
  pull_request:
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,28 +1,30 @@
...
+  push:
+    branches:
+      - main
 
 jobs:
```
Comment (sobolevn) [inline .github/workflows/ci.yml:18]:
```suggestion

concurrency:
  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
  cancel-in-progress: true

permissions:
  contents: read

jobs:
```

**Review by sobolevn — APPROVED** [review]
Thank you! Two optional suggestions.

File: README.md [hunk 3]
```diff
@@ -35,6 +35,8 @@
 Effortlessly build performant APIs with [Litestar](https://litestar.dev/) - The powerful, lightweight and flexible ASGI
 framework
 
+[![Litestar Plugin Architecture](https://img.shields.io/badge/Litestar%20Plugin-%E2%AD%90%20Enabled-202235.svg?logo=python&labelColor=202235&color=edb641&logoColor=edb641)](https://docs.litestar.dev/latest/usage/plugins/index.html) repositories are marked with <sup>*</sup> to indicate they extend functionality through the plugin system.
```
Comment (JacobCoffee) [inline README.md:38]: :(

**Review by JacobCoffee — APPROVED** [review]

**Comment by JacobCoffee** [comment]
probably need to set `$ awesome_bot README.md --allow 403,429` or something to make reddit happy

**Comment by Alc-Alc** [comment]
> probably need to set `$ awesome_bot README.md --allow 403,429` or something to make reddit happy

I thought about this, we already allowed some status codes there but 403 felt like a legit code to not allow, but in this case its rather annoying, I can add 403 if you are ok though

**Comment by JacobCoffee** [comment]
@Alc-Alc maybe awesome-bot allows scoping where the 403 is allowed from

**Pull request merged**